### ISSUE TITLE
(Feature) Setup

### ISSUE THREAD

**Issue by AndreasNasman** [TurkuForge/baby-project-web#1]
### General

- [x] Migrate to [Vite](https://vitejs.dev/)
- [x] Add [Vue Router](https://router.vuejs.org/)
- [x] Add [TailwindCSS](https://tailwindcss.com/) with its [just-in-time compiler](https://github.com/tailwindlabs/tailwindcss-jit)
- [x] Add an `@` alias

### Formatting and linting

- [x] Add [ESLint](https://eslint.org/) with [JavaScript Standard Style](https://standardjs.com/) and [eslint-plugin-vue](https://eslint.vuejs.org/)
- [x] Add [Prettier](https://prettier.io/)
- [x] Add an [EditorConfig](https://editorconfig.org/) file
- [x] Format and lint all files
- [ ] Configure linting, formatting, and Typescript compilation to run on a Git pre-commit hook.
- [ ] Add a GitHub action for linting and formatting.

### PWA
- [x] Add [vite-plugin-pwa](https://github.com/antfu/vite-plugin-pwa)
- [ ] Configure a manifest

### Testing

- [ ] Add [Mocha](https://mochajs.org/) + [Chai](https://www.chaijs.com/) for unit testing
- [ ] Add [Cypress](https://www.cypress.io/) for E2D testing

### Clean up

- [x] Remove all `eslint-disable`s, if possible
- [x] Prefer named exports, potentially in a `index.ts` file

---

#### Disregarded

- [x] ~~Change from `npm` to `yarn`~~ **No advantage of using `yarn`, at least for our use case.**
- [x] ~~Add [stylelint](https://stylelint.io/)~~ **No need (at least initially) since we want to use Tailwind CSS.**
- [x] ~~`index.html` title reference~~ **Seldom you want the `index.html` to be the same as the `name` in `package.json`.**

## Context

It's easier to develop once all this stuff is in place.

## Possible Implementation

Complete all tasks above.

**Comment by dusan-turajlic** [comment]
> Run lint, formatting, and compilation before commits.

Why? I would rather change that to setup github action for linting. I'm not a fan of tasks running when I do my commits

**Comment by dusan-turajlic** [comment]
> Change from npm to yarn

Is there a reason for this? IDK but I have kinda just been sticking with NPM for most of my tasks lately and have not encounterd any issues

**Comment by dusan-turajlic** [comment]
Also should this maybe be split in to a few issues. Some of these can be done in pararell.

**Comment by AndreasNasman** [comment]
> > Run lint, formatting, and compilation before commits.
> 
> Why? I would rather change that to setup github action for linting. I'm not a fan of tasks running when I do my commits

I agree that there should be GitHub action checking this, but I like to catch errors and warnings as soon as possible, so that's why I like having formatting and linting checks on pre-commit Git hooks. Not sure what's the easiest way to have this conveniently be turned on and off according to preferences. 🤔

> > Change from npm to yarn
> 
> Is there a reason for this? IDK but I have kinda just been sticking with NPM for most of my tasks lately and have not encounterd any issues

Not sure why I put this here, `npm` is fine for me too! 👍

> Also should this maybe be split in to a few issues. Some of these can be done in pararell.

I kinda want to do this myself in one go as I'm going to document this process – might put post migration process online somewhere too – and use it at work if possible. 🙂 Most of the items on the checklist are minor issues anyway; connecting them all is a bit more tricky.

**Comment by AndreasNasman** [comment]
Chosen Vue CLI options for reference
![image](https://user-images.githubusercontent.com/17124245/113931386-12a01a80-97fb-11eb-983c-cbe498e86f05.png)
